File: frontend/src/domain/lot.ts
/** Лот — квартира активного набора. Плоская модель: ЖК и адрес корпуса — просто поля. */

export const LOT_STATUSES = ['in_sale', 'booked', 'sold'] as const
export type LotStatus = (typeof LOT_STATUSES)[number]

export const LOT_STATUS_LABELS: Record<LotStatus, string> = {
  in_sale: 'В продаже',
  booked: 'Забронирован',
  sold: 'Продан',
}

export interface Lot {
  id: number
  externalId: string
  setId: number
  project: string
  address: string
  /** 0 — студия */
  rooms: number
  area: number
  floor: number
  price: number
  priceBase: number
  /** Считается бэкендом, не хранится; null, если площадь нулевая */
  pricePerM2: number | null
  status: LotStatus
  createdAt: Date
  updatedAt: Date
}

export const LOT_SORT_FIELDS = ['price', 'price_per_m2', 'area', 'floor', 'rooms'] as const
export type LotSortField = (typeof LOT_SORT_FIELDS)[number]

export type SortOrder = 'asc' | 'desc'

export interface LotQuery {
  project?: string
  rooms?: number[]
  priceM2Min?: number
  priceM2Max?: number
  status?: LotStatus
  sort: LotSortField
  order: SortOrder
  page: number
  pageSize: number
}

export const DEFAULT_LOT_QUERY: LotQuery = {
  sort: 'price',
  order: 'asc',
  page: 1,
  pageSize: 20,
}

File: backend/app/schemas/lot.py
from datetime import datetime
from decimal import Decimal
from enum import StrEnum

from pydantic import Field

from app.domain.enums import LotStatus
from app.schemas.common import ApiModel, PageParams


class LotSortField(StrEnum):
    PRICE = "price"
    PRICE_PER_M2 = "price_per_m2"
    AREA = "area"
    FLOOR = "floor"
    ROOMS = "rooms"


class SortOrder(StrEnum):
    ASC = "asc"
    DESC = "desc"


class LotFilters(PageParams):
    project: str | None = None
    rooms: list[int] | None = Field(None, description="Комнатность, 0 — студия")
    price_m2_min: Decimal | None = Field(None, ge=0)
    price_m2_max: Decimal | None = Field(None, ge=0)
    status: LotStatus | None = None
    sort: LotSortField = LotSortField.PRICE
    order: SortOrder = SortOrder.ASC


class LotOut(ApiModel):
    id: int
    external_id: str
    set_id: int
    project: str
    address: str
    rooms: int
    area: Decimal
    floor: int
    price: Decimal
    price_base: Decimal
    status: LotStatus
    price_per_m2: Decimal | None  # значение — hybrid_property Lot.price_per_m2
    created_at: datetime
    updated_at: datetime
